bluesky/utils.py

```diff
@@ -1336,148 +1336,175 @@ def my_name(obj):
 _qapp = None
 
 
-def default_during_task(blocking_event):
-    """
-    The default setting for the RunEngine's during_task parameter.
+class DuringTask:
 
-    This makes it possible for plots that use matplotlib's Qt backend to update
-    live during data acquisition.
+    def initialize(self):
+        pass
 
-    It solves the problem that Qt must be run from the main thread.
-    If matplotlib and a known Qt binding are already imported, run the
-    matplotlib qApp until the task completes. If not, there is no need to
-    handle qApp: just wait on the task.
-    """
-    global _qapp
-    if 'matplotlib' not in sys.modules:
-        # We are not using matplotlib + Qt. Just wait on the Event.
-        blocking_event.wait()
-    # Figure out if we are using matplotlib with which backend
-    # without importing anything that is not already imported.
-    else:
-        import matplotlib
-        backend = matplotlib.get_backend().lower()
-        # if with a Qt backend, do the scary thing
-        if 'qt' in backend:
-            from matplotlib.backends.qt_compat import QtCore, QtWidgets
-            app = QtWidgets.QApplication.instance()
-            if app is None:
-                _qapp = app = QtWidgets.QApplication([b'bluesky'])
-            assert app is not None
-            event_loop = QtCore.QEventLoop()
-
-            def start_killer_thread():
-                def exit_loop():
-                    blocking_event.wait()
-                    # If the above wait ends quickly, we need to avoid the race
-                    # condition where this thread might try to exit the qApp
-                    # before it even starts.  Therefore, we use QTimer, below,
-                    # which will not start running until the qApp event loop is
-                    # running.
-                    event_loop.exit()
+    def block(self, blocking_event):
+        pass
```

Comment:
```suggestion
        blocking_event.wait()
```

We should make this a usable implementation.